runtime/vm/CMakeLists.txt

```diff
@@ -52,15 +52,21 @@ if(OMR_ARCH_POWER AND OMR_OS_AIX)
 		]]
 	endif()
 elseif(OMR_OS_ZOS)
-	# TODO need to handle optimized debug info
-	list(APPEND interp_flags_to_remove "-O3" "-g" "-qdebug=nohook")
-
-	list(APPEND interp_new_flags
-		-O2
-		"\"-Wc,TBYDBG(-qdebug=MRABIG)\""
-		"\"-Wc,TBYDBG(-qdebug=lincomm:ptranl:tfbagg)\""
-		"\"-Wc,FEDBG(-qxflag=InlineDespiteVolatileInArgs)\""
-	)
+	if(OMR_TOOLCONFIG STREQUAL "openxl")
+		# Hooks are not implemented in Open XL. The default debug compiles in Open XL will be NOHOOK
+		# equivalent.
+		list(APPEND interp_flags_to_remove "-qdebug=nohook")
```

Comment:
> runtime/vm/CMakeLists.txt : Added appropriate conditionals to apply correct flags for Open XL suggested by compiler team. Hooks are not implemented for Open XL.

Just to understand, with xlc it removes `-O3`, `-g` for compiler but with OpenXL it will not ?

Reply:
Why won't we want `-O2` being added for `interp_new_flags` from the XLC case?

Reply:
That might be an oversight, from the discussions I can find right now the intention was only to remove the `"\"-Wc` flags , but leave the same optimizations. So I will try with restoring the optimization levels and rebuilding.
If there is something I may not be remembering, at least it might remind me if it was intentional.

Reply:
I think this is what the intended outcome was, which I am currently testing out ...

```
    if(OMR_TOOLCONFIG STREQUAL "openxl")                                                                                                                                                                                   # Hooks are not implemented in Open XL. The default debug compiles in Open XL will be NOHOOK
        # equivalent.
        list(APPEND interp_flags_to_remove "-O3" "-g")
        list(APPEND interp_new_flags -O2)
    else()
        # TODO need to handle optimized debug info
        list(APPEND interp_flags_to_remove "-O3" "-g" "-qdebug=nohook")

        list(APPEND interp_new_flags
            -O2
            "\"-Wc,TBYDBG(-qdebug=MRABIG)\""
            "\"-Wc,TBYDBG(-qdebug=lincomm:ptranl:tfbagg)\""
            "\"-Wc,FEDBG(-qxflag=InlineDespiteVolatileInArgs)\""
        )    
```

Reply:
Verified these changes are working with openxl, still need to test xlc with jenkins, but I imagine it won't be a problem since that clause/conditional has remained unchanged altogether.